File: Domain/Models/Vehiculos/Marca.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace Domain.Models.Vehiculos
{
    public class Marca
    {
        [Key]
        public int MarcaId { get; set; }

        [Required]
        [Column(TypeName ="nvarchar(200)")]
        public string Descripcion { get; set; }

        public List<Modelo> Modelos { get; set; }

    }
}

File: CarAgency/ClientApp/src/domain/models/vehiculos/Marca.ts
import { IMarca }  from "domain/interfaces/vehiculos/IMarca";
import { Modelo }  from "domain/models/vehiculos/Modelo";

export  class Marca implements IMarca {
    MarcaId? : number | null;
    Descripcion!: string;
    Modelo? : Modelo[];

    public constructor(init?: Partial<Marca>){
        Object.assign(this, init);
    };

}